go/pools/resource_pool.go

```diff
@@ -41,7 +41,7 @@ var (
 	ErrClosed = errors.New("resource pool is closed")
 
 	// ErrTimeout is returned if a resource get times out.
-	ErrTimeout = vterrors.New(vtrpcpb.Code_DEADLINE_EXCEEDED, "resource pool timed out")
+	ErrTimeout = vterrors.New(vtrpcpb.Code_RESOURCE_EXHAUSTED, "resource pool timed out")
```

Comment:
I was thinking that RESOURCE_EXHAUSTED is a better code for this error. LMK if you disagree.